tests/storage/restore.go

```diff
@@ -1083,6 +1089,120 @@ var _ = SIGDescribe("[Serial]VirtualMachineRestore Tests", func() {
+					)
+					if source.Annotations == nil {
+						source.Annotations = make(map[string]string)
+					}
+					source.Annotations["cdi.kubevirt.io/storage.bind.immediate.requested"] = "true"
+
+					source, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(source.Namespace).Create(context.Background(), source, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					sourceDV = source
+					sourceDV = waitDVReady(sourceDV)
+
+					role, roleBinding := tests.GoldenImageRBAC(tests.NamespaceTestAlternative)
+					role, err = virtClient.RbacV1().Roles(role.Namespace).Create(context.TODO(), role, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					cloneRole = role
+					roleBinding, err = virtClient.RbacV1().RoleBindings(roleBinding.Namespace).Create(context.TODO(), roleBinding, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					cloneRoleBinding = roleBinding
+				})
+
+				AfterEach(func() {
+					if sourceDV != nil {
+						err := virtClient.CdiClient().CdiV1beta1().DataVolumes(sourceDV.Namespace).Delete(context.TODO(), sourceDV.Name, metav1.DeleteOptions{})
+						Expect(err).ToNot(HaveOccurred())
+					}
+
+					if cloneRole != nil {
+						err := virtClient.RbacV1().Roles(cloneRole.Namespace).Delete(context.TODO(), cloneRole.Name, metav1.DeleteOptions{})
+						Expect(err).ToNot(HaveOccurred())
+					}
+					if cloneRoleBinding != nil {
+						err = virtClient.RbacV1().RoleBindings(cloneRoleBinding.Namespace).Delete(context.TODO(), cloneRoleBinding.Name, metav1.DeleteOptions{})
+						Expect(err).ToNot(HaveOccurred())
+					}
+				})
+
+				checkCloneAnnotations := func(vm *v1.VirtualMachine, shouldExist bool) {
+					pvcName := ""
+					for _, v := range vm.Spec.Template.Spec.Volumes {
+						n := typesutil.PVCNameFromVirtVolume(&v)
+						if n != "" {
+							Expect(pvcName).Should(Equal(""))
+							pvcName = n
+						}
+					}
+					pvc, err := virtClient.CoreV1().PersistentVolumeClaims(vm.Namespace).Get(context.TODO(), pvcName, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					for _, a := range []string{"k8s.io/CloneRequest", "k8s.io/CloneOf"} {
+						_, ok := pvc.Annotations[a]
+						Expect(ok).Should(Equal(shouldExist))
+					}
+				}
+
+				createVMFromSource := func() *v1.VirtualMachine {
+					dataVolume := tests.NewRandomDataVolumeWithPVCSource(
+						sourceDV.Namespace,
+						sourceDV.Name,
+						util.NamespaceTestDefault,
+						corev1.ReadWriteOnce,
+					)
+					tests.SetDataVolumePVCStorageClass(dataVolume, snapshotStorageClass)
+					tests.SetDataVolumePVCSize(dataVolume, "6Gi")
+					vmi := tests.NewRandomVMIWithDataVolume(dataVolume.Name)
+					tests.AddUserData(vmi, "cloud-init", tests.BashHelloScript)
+					vm := tests.NewRandomVirtualMachine(vmi, false)
+					tests.AddDataVolumeTemplate(vm, dataVolume)
+					return vm
+				}
+
+				It("should restore a vm that boots from a network cloned datavolumetemplate to the same VM", func() {
```

Comment:
Am I correct in assuming that this is the test that fails?  As long as the following one passes, I am okay with deleting this one.  The code change is still exercised by following test.

Reply:
Yes, VM with datavolume works on older CDI, datavolume template only on the with the fix. 

And you are right, here we only care if the original problem was solved. Both tests pass on main, so I'll delete this one and rerun.